bob/src/cluster/operations.rs

```diff
@@ -295,7 +276,48 @@ pub(crate) async fn put_local_node(
     backend.put_local(key, data, op).await
 }
 
+// =================== EXIST ==================
 
+pub(crate) async fn exist_on_local_node(
+    backend: &Backend,
+    keys: &[BobKey],
+) -> Result<Vec<bool>, Error> {
+    Ok(backend
+        .exist(keys, &BobGetOptions::new_get(Some(GetOptions::new_local())))
+        .await?)
+}
+
+pub(crate) async fn exist_on_local_alien(
+    backend: &Backend,
+    keys: &[BobKey],
+) -> Result<Vec<bool>, Error> {
+    Ok(backend
+        .exist(keys, &BobGetOptions::new_get(Some(GetOptions::new_alien())))
+        .await?)
+}
+
+pub(crate) async fn exist_on_remote_nodes(
+    nodes: &[&Node],
```

Comment:
Looks like `nodes` is not needed. You can take `Node` instance from `keys_by_node`: `keys_by_node.values().map(|n| &n.0)`